src/loaders/compute_tools/microtrait/microtrait.py

```diff
@@ -47,6 +75,53 @@ def _r_table_to_df(r_table):
     return df
 
 
+def _process_trait_counts(
+        trait_counts_df: pd.DataFrame,
+        data_id: str):
+    # process the trait counts file to create the heatmap data and metadata
+
+    trait_df = trait_counts_df[_MICROTRAIT_TO_SYS_TRAIT_MAP.keys()]
+
+    # Check if the trait index column has non-unique values
+    if len(trait_df[loader_common_names.MICROTRAIT_TRAIT_NAME].unique()) != len(trait_df):
+        raise ValueError(f"The {loader_common_names.MICROTRAIT_TRAIT_NAME} column has non-unique values")
+
+    # Extract the substring of the 'microtrait_trait-displaynamelong' column before the first colon character
+    # and assign it to a new 'category' column in the DataFrame
+    trait_df[loader_common_names.SYS_TRAIT_CATEGORY] = \
+    trait_df[loader_common_names.MICROTRAIT_TRAIT_DISPLAYNAME_LONG].str.split(':').str[0]
```

Comment:
```suggestion
    trait_df[loader_common_names.SYS_TRAIT_CATEGORY] = trait_df[
        loader_common_names.MICROTRAIT_TRAIT_DISPLAYNAME_LONG].str.split(':').str[0]
```

Although if the constants move back here that'll be moot

Reply:
👍

Reply:
:+1: